storage/index/block.go

```diff
@@ -87,8 +95,13 @@ func NewBlock(
 	}
 
 	b := &block{
-		state:     blockStateOpen,
-		segment:   seg,
+		state:         blockStateOpen,
+		needsFlush:    true,
```

Comment:
Can we make this a computed function that returns true/false?  i.e. inspecting inactiveMutableSegments and the activeSegment, etc.

Reply:
yeah good call, makes the code a lot simpler

Reply:
done